app/pages/project/vpcs/VpcPage/tabs/VpcGatewaysTab.tsx

```diff
@@ -0,0 +1,179 @@
+import { useMemo } from 'react'
+import { Link, Outlet, type LoaderFunctionArgs } from 'react-router-dom'
+
+import { apiq, getListQFn, queryClient, type InternetGateway } from '~/api'
+import { getVpcSelector, useVpcSelector } from '~/hooks/use-params'
+import { EmptyCell } from '~/table/cells/EmptyCell'
+import { IpPoolCell } from '~/table/cells/IpPoolCell'
+import { makeLinkCell } from '~/table/cells/LinkCell'
+import { Columns } from '~/table/columns/common'
+import { useQueryTable } from '~/table/QueryTable'
+import { CopyableIp } from '~/ui/lib/CopyableIp'
+import { EmptyMessage } from '~/ui/lib/EmptyMessage'
+import { ALL_ISH } from '~/util/consts'
+import { pb } from '~/util/path-builder'
+import type * as PP from '~/util/path-params'
+
+const gatewayList = ({ project, vpc }: PP.Vpc) =>
+  getListQFn('internetGatewayList', { query: { project, vpc, limit: ALL_ISH } })
+const routerList = ({ project, vpc }: PP.Vpc) =>
+  getListQFn('vpcRouterList', { query: { project, vpc, limit: ALL_ISH } })
+const routeList = ({ project, vpc, router }: PP.VpcRouter) =>
+  getListQFn('vpcRouterRouteList', { query: { project, vpc, router, limit: ALL_ISH } })
+const gatewayIpAddressList = ({ gatewayId }: { gatewayId: string }) =>
+  getListQFn('internetGatewayIpAddressList', { query: { gateway: gatewayId } })
+const gatewayIpPoolList = ({ gatewayId }: { gatewayId: string }) =>
+  getListQFn('internetGatewayIpPoolList', { query: { gateway: gatewayId } })
+const projectIpPoolList = getListQFn('projectIpPoolList', { query: { limit: ALL_ISH } })
+
+const InternetGatewayIpAddressCell = ({ gatewayId }: { gatewayId: string }) => {
+  const { data: addresses } = useQuery(gatewayIpAddressList({ gatewayId }).optionsFn())
+  if (!addresses || addresses.items.length < 1) return <EmptyCell />
+  return <CopyableIp ip={addresses.items[0].address} isLinked={false} />
+}
+
+const InternetGatewayIpPoolCell = ({ gatewayId }: { gatewayId: string }) => {
+  const { data: gateways } = useQuery(gatewayIpPoolList({ gatewayId }).optionsFn())
+  if (!gateways || gateways.items.length < 1) return <EmptyCell />
+  return <IpPoolCell ipPoolId={gateways.items[0].ipPoolId} />
+}
+
+// called by InternetGatewayAttachedRoutesCell to get the routes per router
+// we need to have this in its own function because useQuery cannot be called inside a loop
+const InternetGatewayRoutes = ({
+  project,
+  vpc,
+  gateway,
+  router,
+}: PP.VpcInternetGateway & { router: string }) => {
+  const { data: routes } = useQuery(routeList({ project, vpc, router }).optionsFn())
+  if (!routes || routes.items.length < 1) return null
+  return routes.items
+    .filter((r) => r.target.type === 'internet_gateway' && r.target.value === gateway)
+    .map((route) => (
+      <Link
+        key={route.name}
+        to={pb.vpcRouterRouteEdit({ project, vpc, router, route: route.name })}
+        className="link-with-underline text-sans-md"
+      >
+        {route.name}
+      </Link>
```

Comment:
I find clicking this link to the route edit form very jarring. We have some similar links in other places (linked silos and IP pools both link to each other, disks link to the instances they're attached to) but somehow this feels much more confusing. I thought maybe the problem was being dropped in the form (because you're in a modal and can't see much behind it), so maybe landing in the list of routes would be better, but landing on the router route list page instead is similarly disorienting. This isn't really something wrong with this PR (it's more about the underlying issue of deep nesting in the VPC tree, which is why @benjaminleonard wanted to keep the gateway details in a modal instead of a whole page) so I'm not sure we can fix it here either, unless we were to drop the link altogether. A more elaborate fix would be to open the route details in a side modal on top of the gateways page, but it's never great to have two canonical locations for a thing.

Reply:
Did we figure out a ballpark likelihood of there being multiple routes targeting a gateway? It doesn't seem like that weird a thing to do. In any case we need to figure out how to render this better as a list — there are 3 here and no spaces between them. I think this is why we were considering not putting this in the cell at all.

<img width="1038" alt="image" src="https://github.com/user-attachments/assets/af62efdd-23ed-491d-9a6f-0fad19d51df6" />